From bartoldeman@users.sourceforge.net Sat Jul  1 20:15:37 2006
Date: Sun, 2 Jul 2006 15:07:10 +1200 (NZST)
From: Bart Oldeman <bartoldeman@users.sourceforge.net>
To: greg@kroah.com, rodent@rodent.za.net
Subject: USB: ipw.c driver fix
Message-ID: <Pine.LNX.4.64.0607021457200.3559@enm-bo-lt.localnet>

The below patch fixes the ipw module in kernel 2.6.17 for me; without
this change it simply does not work at all (all but the first writes are
refused because write_urb_busy is always 1).

This problem was there in 2.6.15 as well, but at that point I used the
(updated) ipw.c, version 0.4, from
http://www.neology.co.za/products/opensource/ipwireless/ which no longer
compiles with 2.6.17. It can be made to after a few changes but
obviously it's easier if the built-in ipw driver works instead of having
to download one from the neology site.

From: Bart Oldeman <bartoldeman@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/ipw.c |    2 ++
 1 file changed, 2 insertions(+)

--- gregkh-2.6.orig/drivers/usb/serial/ipw.c
+++ gregkh-2.6/drivers/usb/serial/ipw.c
@@ -373,6 +373,8 @@ static void ipw_write_bulk_callback(stru
 
 	dbg("%s", __FUNCTION__);
 
+	port->write_urb_busy = 0;
+
 	if (urb->status)
 		dbg("%s - nonzero write bulk status received: %d", __FUNCTION__, urb->status);
 
